src/frontend/src/optimizer/mod.rs

```diff
@@ -158,8 +158,25 @@ impl PlanRoot {
         }
     }
 
+    /// Check whether the plan has `LogicalApply` that isn't be unnested.
+    pub fn has_apply(plan: PlanRef) -> bool {
+        pub struct PlanApplyFinder {
+            has_apply: bool,
+        }
+
+        impl PlanVisitor<()> for PlanApplyFinder {
+            fn visit_logical_apply(&mut self, _plan: &LogicalApply) {
+                self.has_apply = true;
+            }
+        }
```

Comment:
```suggestion
        pub struct PlanApplyFinder {}

        impl PlanVisitor<bool> for PlanApplyFinder {
            fn visit_logical_apply(&mut self, _plan: &LogicalApply) -> bool {
                true
            }
        }
```

Reply:
See #5058 @xxchan . This modification is not equivalent.